### PR TITLE
Allow users to edit files with Gitpod.

### LINKED ISSUE #21: Allow to edit files and then create a pull request
Or at least export a patch file or something. That would be really cool 😀
Awesome project!

### LINKED ISSUE #107: Terminal Integration
First of all, thanks for making this amazing project. It is helpful and easy to use. 
The only thing that I found out missing is the terminal. Is it possible to add the terminal and features to update and commit directly from here?

### LINKED ISSUE #138: Can I run js code online?

### LINKED ISSUE #156: Support for GitHub Pull Request/Commit
We should support for PR/Commit next to provide a better CodeReview experience

https://code.visualstudio.com/api/extension-guides/scm-provider

https://docs.github.com/en/rest/reference/pulls

### LINKED ISSUE #180: option to sign in and edit code w/ full vscode features

### LINKED ISSUE #213: Integrate with Gitpod to develop in the cloud
Similarly to how `github1s` can be developed on [Gitpod](https://www.gitpod.io) (see "Cloud-based development" in the [`README.md`](https://github.com/conwnet/github1s#cloud-based-development), allow anyone to open their project in Gitpod to edit files, open PRs, start Docker containers, etc.

Fixes:
* #180
* #156
* #138
* #107
* #21

### PR DESCRIPTION
**Pull request by mikenikles**
Fixes #213, fixes #180, fixes #156, fixes #138, fixes #107, fixes #21

As discussed with @conwnet, this PR adds a button to the notification at the bottom of the screen that opens a project in [Gitpod](https://www.gitpod.io/).

With Gitpod, you get an automated development environment where you can edit files, run Docker containers, create pull requests and more.

### CODE REVIEW

File: src/vs/github1s/notification.ts
Comment (xcv58) [inline src/vs/github1s/notification.ts:12]: On the deploy https://github1s-git-fork-mikenikles-integrate-with-gitpod-4fd92e.vercel.app/
It gives me the same URL as the current URL but not the gitpod link.

image

Comment (mikenikles) [reply src/vs/github1s/notification.ts:12]: That's because the development environment starts without a repository in the URL. If you open https://github1s-git-fork-mikenikles-integrate-with-gitpod-4fd92e.vercel.app/conwnet/github1s, the button link is ok.

From what I can tell, github1s manually loads `conwnet/github1s` if no project is specified in the URL. Do you think it makes sense to update the URL to `/conwnet/github1s` in the development environment if no other project is specified?

Comment (xcv58) [reply src/vs/github1s/notification.ts:12]: Got it, I think it's ok to leave it as it is.

File: src/vs/github1s/notification.ts
Comment (xcv58) [inline src/vs/github1s/notification.ts:25]: I suggest we use a different way to promote the link since the notification will be invisible after users click the `OK` button.
image

There are a couple of places I can imagine:
- The left bar 
image

- The command line interface 
image

Comment (mikenikles) [reply src/vs/github1s/notification.ts:25]: I agree, the notification bar was the lowest hanging fruit 😄 . I like your suggestions, let me look into that tomorrow (it's getting late 😴 ) and update the PR. Unless of course you feel like working on the PR before I wake up :).

Thanks for your support and feedback, I appreciate it!

Comment (mikenikles) [reply src/vs/github1s/notification.ts:25]: I added a command in my dev environment.

Any idea if there's a way to trust the domain by default? I already added `*.gitpod.io` to `src/vs/workbench/contrib/url/browser/trustedDomains.ts` (and also tried with `gitpod.io`).

image

Comment (xcv58) [reply src/vs/github1s/notification.ts:25]: You can add trust domain here: https://github.com/conwnet/github1s/blob/master/src/vs/workbench/contrib/url/browser/trustedDomains.ts#L304

Comment (xcv58) [reply src/vs/github1s/notification.ts:25]: I tried the command on https://github1s-git-fork-mikenikles-integrate-with-gitpod-4fd92e.vercel.app/conwnet/github1s/blob/HEAD/tsconfig.json and it works without the external website warning.

Comment (mikenikles) [reply src/vs/github1s/notification.ts:25]: I've just pushed 80df292 which does require the trusted domain the same way it currently happens in `sponsors.ts`.

Thanks so much for your hints and guidance! I decided to use the status bar rather than the left bar. We can see how this goes and decide whether an additional left side icon will be required. Does that sound like a plan?

Comment (xcv58) [reply src/vs/github1s/notification.ts:25]: Sounds great, thank you very much!

File: README.md
Comment (xcv58) [inline README.md:29]: Do you mind adding one or two screenshots here? We could do it late if it's inconvenient for you.

**Review by xcv58 — APPROVED** [review]
Tested on https://github1s-git-fork-mikenikles-integrate-with-gitpod-4fd92e.vercel.app/

It looks great, appreciate your contribution 🙏

File: README.md
Comment (mikenikles) [reply README.md:29]: For sure, I'll get to it a bit later today 👍

**Comment by mikenikles** [comment]
@xcv58 or @conwnet this is ready to be merged from my side. I'll let one of you press the merge button :)

**Review by conwnet — APPROVED** [review]
thanks for your work~

**Pull request merged**
